src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -898,5 +898,18 @@ internal static bool IsPunctuatorColonNextToCursor(int cursorPos, string script)
             var colonLen = TexlLexer.PunctuatorColon.Length;
             return script.Length >= cursorPos + colonLen && script.Substring(cursorPos, colonLen) == TexlLexer.PunctuatorColon;
         }
+
+        internal static bool ContainsDataEntityType(DType type, int maxDepth, int currentDepth = 1)
```

Comment:
>int maxDepth, int currentDepth = 1 [](http://example.com/codeflow?start=64&length=34)

It's often simpler to have depth count down towards 0. (rather than count up and pass a maxdepth)